### ISSUE TITLE
Sparse Attention Slower than Flash Attention on Single A100

### ISSUE THREAD

**Issue by OldZhao** [BienLuky/Rectified-SpaAttn#1]
Environment
GPU: single NVIDIA A100 80 GB
CUDA 12.4, PyTorch 2.6, Triton 3.2

DiT One Iteration Time:
python scripts/main_hunyuan.py  --> 76.98s
python scripts/main_hunyuan.py --mode flash  -->72s

Please help me check why sparse is slower, thank you.

**Comment by BienLuky** [comment]
Sparse attention methods are typically designed for long-sequence scenarios. However, when the token sequence is relatively short (e.g., only a few thousand tokens), the overhead of generating sparse attention masks introduces non-negligible latency. As a result, even though fewer tokens participate in the attention computation, the overall attention latency remains largely unchanged.

**Comment by OldZhao** [comment]
I am using the Hunyuan demo in this project. The video specifications are 720p, 128 frames. The token shape is  (1, 24, 115456, 128).  This token seems relatively long. Why am I still not seeing any acceleration?

**Comment by BienLuky** [comment]
Hi, could you provide additional details about your usage? In particular, it would be helpful to know which parts of the project you modified, as well as the inference times for both sparse attention and no sparse FlashAttention. From the current information, the token sequence length appears to be normal.

**Comment by 2302664592** [comment]
What is the minimum resolution required for image generation to achieve a speedup?

**Comment by BienLuky** [comment]
> What is the minimum resolution required for image generation to achieve a speedup?

In our default setting, the generated image resolution is 4096 × 4096, as we set the upscaling factor to 4.
